backend/common/src/main/java/ai/verta/modeldb/common/authservice/RoleServiceUtils.java

```diff
@@ -140,56 +141,51 @@ public boolean deleteEntityResourcesWithServiceUser(
-                            == modelDBServiceResourceTypes
-                        && item.getService() == Service.MODELDB_SERVICE)
-            .findFirst();
-    if (responseItem.isPresent()) {
-      return responseItem.get();
-    }
-    throw new IllegalArgumentException(
-        "Failed to locate "
-            + modelDBServiceResourceTypes.name()
-            + " resources in UAC for "
-            + modelDBServiceResourceTypes.name()
-            + " ID "
-            + entityId);
-  }
-
-  public List<GetResourcesResponseItem> getResourceItems(Optional<Resources> filterTo) {
+      Optional<String> entityId, Optional<String> entityName, Optional<String> workspaceName, ModelDBServiceResourceTypes modelDBServiceResourceTypes) {
     try (AuthServiceChannel authServiceChannel = getAuthServiceChannel()) {
-      final GetResources.Builder getResourcesBuilder = GetResources.newBuilder();
-      filterTo.ifPresent(getResourcesBuilder::setResources);
+      ResourceType resourceType =
+              ResourceType.newBuilder()
+                      .setModeldbServiceResourceType(modelDBServiceResourceTypes)
+                      .build();
+      Resources.Builder resources =
+              Resources.newBuilder()
+                      .setResourceType(resourceType)
+                      .setService(Service.MODELDB_SERVICE);
+      entityId.ifPresent(resources::addResourceIds);
+
+      final GetResources.Builder getResourcesBuilder = GetResources.newBuilder().setResources(resources);
+      entityName.ifPresent(getResourcesBuilder::setResourceName);
+      workspaceName.ifPresent(getResourcesBuilder::setWorkspaceName);
 
       final GetResources.Response response =
-          authServiceChannel
-              .getCollaboratorServiceBlockingStub()
-              .getResources(getResourcesBuilder.build());
-      return response.getItemList();
+              authServiceChannel
+                      .getCollaboratorServiceBlockingStub()
+                      .getResources(getResourcesBuilder.build());
+      List<GetResourcesResponseItem> responseItems = response.getItemList();
+      if (responseItems.size() > 1) {
+        LOGGER.warn(
+                "Role service returned {}"
+                        + " resource response items fetching {} resource, but only expected 1. ID: {}",
+                responseItems.size(),
+                modelDBServiceResourceTypes.name(),
+                entityId);
+      }
+      Optional<GetResourcesResponseItem> responseItem = responseItems.stream().findFirst();
+      if (responseItem.isPresent()){
+        return responseItem.get();
+      } else {
+        StringBuilder errorMessage =
+                new StringBuilder("Failed to locate ")
+                        .append(modelDBServiceResourceTypes.name())
+                        .append(" resources in UAC for ")
+                        .append(modelDBServiceResourceTypes.name());
```

Comment:
I don't think you meant to use the same `modelDBServiceResourceTypes.name()` in both places.

Reply:
No, it is the expected and added intentionally. the error message generated by it will be like 
`Failed to locate PROJECT resources in UAC for PROJECT Name projectName` 
OR
`Failed to locate PROJECT resources in UAC for PROJECT Id abcd-abcd-abcd-abcd`

Reply:
let's just skip the second one then

Reply:
No we can not skip anyone because we have that both get by name and get by Id So both the case will works. look the line diff bellow this comment on file changes tab.